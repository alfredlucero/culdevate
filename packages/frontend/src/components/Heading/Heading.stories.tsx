import React from "react";
import { storiesOf } from "@storybook/react";
import Heading from "./index";

storiesOf("Common/Heading", module)
  .add("H1 Variant", () => <Heading variant="h1">H1 Variant</Heading>)
  .add("H1 Italic", () => (
    <Heading variant="h1" italic={true}>
      H1 Italic
    </Heading>
  ))
  .add("H2 Variant", () => <Heading variant="h2">H2 Variant</Heading>)
  .add("H3 Variant", () => <Heading variant="h3">H3 Variant</Heading>)
  .add("H4 Variant", () => <Heading variant="h4">H4 Variant</Heading>);
