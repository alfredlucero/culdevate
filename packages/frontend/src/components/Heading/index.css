.heading-1 {
  @apply text-4xl;
}

.heading-2 {
  @apply text-3xl;
}

.heading-3 {
  @apply text-2xl;
}

.heading-4 {
  @apply text-xl;
}

.heading-italic {
  @apply italic;
}
