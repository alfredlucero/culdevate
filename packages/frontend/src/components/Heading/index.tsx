import React from "react";
import cn from "classnames";
import { CommonProps } from "../commonProps";

interface HeadingProps extends CommonProps {
  variant: HeadingVariant;
  children: React.ReactNode;
  italic?: boolean;
}

type HeadingVariant = "h1" | "h2" | "h3" | "h4";

const headingBaseClasses = ["font-mono", "font-bold", "text-gray-800"];

const Heading: React.FC<HeadingProps> = ({
  variant,
  children,
  italic = false,
  testId = "",
  className = "",
  ...passThroughProps
}) => {
  if (variant === "h1") {
    return (
      <h1
        className={`${cn(headingBaseClasses, "text-4xl", {
          "heading-italic": italic,
        })} ${className}`}
        {...(testId !== "" ? { "data-testid": testId } : {})}
        {...passThroughProps}
      >
        {children}
      </h1>
    );
  }

  if (variant === "h2") {
    return (
      <h2
        className={`${cn(headingBaseClasses, "text-3xl", {
          italic: italic,
        })} ${className}`}
        {...(testId !== "" ? { "data-testid": testId } : {})}
        {...passThroughProps}
      >
        {children}
      </h2>
    );
  }

  if (variant === "h3") {
    return (
      <h3
        className={`${cn(headingBaseClasses, "text-2xl", {
          italic: italic,
        })} ${className}`}
        {...(testId !== "" ? { "data-testid": testId } : {})}
        {...passThroughProps}
      >
        {children}
      </h3>
    );
  }

  if (variant === "h4") {
    return (
      <h4
        className={`${cn(headingBaseClasses, "text-xl", {
          italic: italic,
        })} ${className}`}
        {...(testId !== "" ? { "data-testid": testId } : {})}
        {...passThroughProps}
      >
        {children}
      </h4>
    );
  }

  return null;
};

export default Heading;
