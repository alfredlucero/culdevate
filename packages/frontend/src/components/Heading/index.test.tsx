import React from "react";
import { render } from "@testing-library/react";
import Heading from "./index";

describe("<Heading />", () => {
  const headingTestId = "headingTestId";

  test("should render header one variant without error", () => {
    const { container } = render(
      <Heading variant="h1" testId={headingTestId} className="extra-heading-class">
        Header One Content
      </Heading>,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render header two variant without error", () => {
    const { container } = render(
      <Heading variant="h2" testId={headingTestId} className="extra-heading-class">
        Header Two Content
      </Heading>,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render header three variant without error", () => {
    const { container } = render(
      <Heading variant="h3" testId={headingTestId} className="extra-heading-class">
        Header Three Content
      </Heading>,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render header four variant without error", () => {
    const { container } = render(
      <Heading variant="h4" testId={headingTestId} className="extra-heading-class">
        Header Four Content
      </Heading>,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render italic class when italic", () => {
    const { getByTestId } = render(
      <Heading variant="h1" testId={headingTestId} italic={true}>
        Header One Content
      </Heading>,
    );

    expect(getByTestId(headingTestId)).toHaveClass("heading-italic");
  });
});
