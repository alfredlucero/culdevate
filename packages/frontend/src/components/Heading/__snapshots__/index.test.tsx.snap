// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render header four variant without error 1`] = `
<div>
  <h4
    class="font-mono font-bold text-gray-800 text-xl extra-heading-class"
    data-testid="headingTestId"
  >
    Header Four Content
  </h4>
</div>
`;

exports[`<Heading /> should render header one variant without error 1`] = `
<div>
  <h1
    class="font-mono font-bold text-gray-800 text-4xl extra-heading-class"
    data-testid="headingTestId"
  >
    Header One Content
  </h1>
</div>
`;

exports[`<Heading /> should render header three variant without error 1`] = `
<div>
  <h3
    class="font-mono font-bold text-gray-800 text-2xl extra-heading-class"
    data-testid="headingTestId"
  >
    Header Three Content
  </h3>
</div>
`;

exports[`<Heading /> should render header two variant without error 1`] = `
<div>
  <h2
    class="font-mono font-bold text-gray-800 text-3xl extra-heading-class"
    data-testid="headingTestId"
  >
    Header Two Content
  </h2>
</div>
`;
